src/store.ts

```diff
@@ -0,0 +1,36 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+import { User } from 'firebase/app';
+
+import * as fb from './firebaseConfig';
+
+Vue.use(Vuex);
+
+type State = {
+  currentUser: User | null;
+  userProfile: {};
+};
+
+const store = new Vuex.Store<State>({
+  state: {
+    currentUser: null,
+    userProfile: {}
+  },
+  actions: {
+    fetchUserProfile({ commit, state }) {}
+  },
+  mutations: {
+    setCurrentUser(state: State, val: User) {
+      state.currentUser = val;
+    }
+  }
+});
```

Comment:
What's the purpose of using Vuex.Store in this case?

Reply:
@SamChou19815 do you mind adding some comments on store.ts to explain what it does? I don't have much experience with it and I think it could help the devs as well

Reply:
See the deleted `src/store.js`. The original file also uses Vuex.Store, and I'm not sure why.

Reply:
I guess it's like redux in the React world. However, I think currently it's being misused. I think using it storing all the courses taken would be a better idea for using the store.

Reply:
Interesting... I'll make a note to ask Will and the devs about it